src/full/Agda/TypeChecking/Rewriting.hs

```diff
@@ -194,6 +194,15 @@ checkRewriteRule q = do
   -- for a type signature whose body has not been type-checked yet.
   when (isEmptyFunction $ theDef def) $
     typeError $ IllegalRewriteRule q BeforeFunctionDefinition
+  -- Issue 6643: Also check that there are no mututal definitions
+  -- that are not yet defined.
+  whenJustM (asksTC envMutualBlock) $ \ mb -> do
```

Comment:
NB: the `$` before `\` is no longer needed thanks to `BlockArguments`.

Reply:
Oh, that's some muscle memory that is going to take a while to unlearn :)